### PR TITLE
fix project

### PR DESCRIPTION
**Pull request by Aliaksandrzaz**

### CODE REVIEW

File: js/src/features/project/components/Project/ProjectCreateForm.tsx [hunk 1]
```diff
@@ -49,24 +45,27 @@ export const CreateProjectForm: React.FC<Props> = ({
...
-      onFinish={onFinish}
-      onFinishFailed={onFinishFailed}
-    >
+    <Form form={form} labelCol={{ span: 8 }} onFinish={onFinish}>
       <Form.Item
         label="Name"
         name="name"
-        rules={[{ required: true, message: "Please input Name!" }]}
+        rules={[
+          { required: true, message: "Please input name" },
```
[the remark this answers is not in the document]
Comment (kwoon) [reply js/src/features/project/components/Project/ProjectCreateForm.tsx:53]: For required error you can do something like this

```ts
const getRequiredErrors = (min: number, trim = true) => {
    const required = { required: true, message: "Required" };
    const whitespace = { whitespace: true, message: "Required" };
    return trim ? [required, whitespace] : [required];
];
```

And use it

```ts
rules={[...getRequiredErrors()]}
```

File: js/src/features/project/components/Project/ProjectSlugForm.tsx [hunk 2]
```diff
@@ -41,25 +41,18 @@ export const ProjectSlugForm: React.FC<Props> = ({
...
-      onFinishFailed={onFinishFailed}
-      initialValues={initialValues}
-    >
+    <Form form={form} onFinish={onFinish} initialValues={initialValues}>
       <Form.Item
         name="slug"
         rules={[
-          { required: true, message: "Please input Slug!" },
+          { required: true, message: "Please input slug" },
           {
```
Comment (kwoon) [inline js/src/features/project/components/Project/ProjectSlugForm.tsx:50]: This validation you can unify to `getUrlError`

File: js/src/features/project/components/Project/ProjectSlugForm.tsx [hunk 3]
```diff
@@ -41,25 +41,18 @@ export const ProjectSlugForm: React.FC<Props> = ({
...
             pattern: /^[a-z0-9]+(?:[-_][a-z0-9]+)*$/,
-            message: "Please input valid Slug!",
+            message: "Please input valid slug",
```
Comment (kwoon) [inline js/src/features/project/components/Project/ProjectSlugForm.tsx:52]: Message can be just `Not valid`

File: js/src/features/project/components/Role/RoleFormTableWrapper.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ type Props = {
   permissions: string[];
 }
 
-export const ProjectRoleFormTableWrapper: React.FC<Props> = ({
+export const RoleFormTableWrapper: React.FC<Props> = ({
```
Comment (kwoon) [inline js/src/features/project/components/Role/RoleFormTableWrapper.tsx:7]: It is not wrapper. Just `RoleFormTable`

File: js/src/features/project/epics.ts [hunk 5]
```diff
@@ -23,7 +24,8 @@ import {
   addProjectTeamMemberAsync,
   closeProjectTeamNewMemberFormAction,
 } from "./actions";
-import { push } from "connected-react-router";
+import { notificationError } from "../notification/actions";
+import { SOMETHING_WENT_WRONG } from "../../constants";
```
Comment (kwoon) [inline js/src/features/project/epics.ts:28]: Better to put in notifications like default error notification (if you don't pass any argument)

```ts
dispatch(notificate({ type: 'error' }) // the same as `dispatch(notificate({ type: 'error', message: SOMETHING_WENT_WRONG })` 
```

File: js/src/services/team-api.ts [hunk 6]
```diff
@@ -78,7 +78,7 @@ export const changeStatus = ({
   api
     .post(`/teams/${teamId}/${status}`, {})
     .then<Success<null>>(() => ({ success: true, data: null }))
-    .catch<FormFailure>((resp) => ({
+    .catch<Failure>((resp) => ({
       success: false,
-      errors: resp?.response?.data?.errors || [],
+      reason: "Something went wrong",
```
Comment (kwoon) [inline js/src/services/team-api.ts:83]: Reason should be like a constant or variable name, e.g. 

```ts
reason: "something_went_wrong"
```

File: js/src/services/project-api.ts [hunk 7]
```diff
@@ -31,7 +31,7 @@ export const fetchProject = (slug: string) => {
       success: true,
       data: resp.data,
     }))
-    .catch<Failure>(() => ({ success: false, reason: "Error" }));
+    .catch<Failure>(() => ({ success: false, reason: "something_wrong" }));
```
Comment (kwoon) [inline js/src/services/project-api.ts:34]: Your `Something went wrong` reason everywhere the different.

File: js/src/pages/ProjectsPage.tsx [hunk 8]
```diff
@@ -11,12 +11,12 @@ import {
...
-  selectIsLoading,
-  selectErrors,
   selectProjects,
+  selectCreateProjectFormErrors,
+  selectIsLoadingProjects,
 } from "../features/project/selectors";
-import { ProjectList } from "../features/project/components/ProjectList";
-import { CreateProjectForm } from "../features/project/components/ProjectCreateForm";
+import { ProjectList } from "../features/project/components/Project/ProjectList";
+import { CreateProjectForm } from "../features/project/components/Project/ProjectCreateForm";
```
Comment (kwoon) [inline js/src/pages/ProjectsPage.tsx:19]: ProjectCreateForm => NewProjectForm

File: js/src/pages/ProjectsPage.tsx [hunk 9]
```diff
@@ -44,8 +44,8 @@ export const ProjectsPage: React.FC = () => {
     [dispatch],
   );
 
-  const errors = useSelector(selectErrors);
-  const loading = useSelector(selectIsLoading);
+  const createFormErrors = useSelector(selectCreateProjectFormErrors);
```
Comment (kwoon) [inline js/src/pages/ProjectsPage.tsx:47]: createFormErrors => newFormErrors
selectCreateProjectFormErrors => selectNewProjectFormErrors

File: js/src/pages/ProjectTeamsPage.tsx [hunk 10]
```diff
@@ -75,9 +73,9 @@ export const ProjectTeamsPage: React.FC = () => {
         footer={null}
         destroyOnClose={true}
       >
-        <ProjectTeamsCreateForm
+        <TeamCreateForm
```
Comment (kwoon) [inline js/src/pages/ProjectTeamsPage.tsx:76]: TeamCreateForm => NewTeamForm

File: js/src/pages/ProjectTeamsPage.tsx [hunk 11]
```diff
@@ -46,10 +46,8 @@ export const ProjectTeamsPage: React.FC = () => {
   }, [dispatch]);
 
   const project = useSelector(selectProject);
-  const errors = useSelector(selectErrors);
-  const isOpenTeamCreateForm = useSelector(
-    selectIsOpenCreateTeamForm,
-  );
+  const newTeamFormErrors = useSelector(selectNewTeamFormErrors);
+  const isOpenTeamCreateForm = useSelector(selectIsOpenCreateTeamForm);
```
Comment (kwoon) [inline js/src/pages/ProjectTeamsPage.tsx:50]: isOpenTeamCreateForm => isOpenNewTeamForm

File: js/src/pages/ProjectTeamPage.tsx [hunk 12]
```diff
@@ -92,8 +93,11 @@ export const ProjectTeamPage: React.FC = () => {
 
   const team = useSelector(selectProjectTeam);
   const project = useSelector(selectProject);
-  const isOpenTeamNewMemberForm = useSelector(selectIsOpenTeamAddMemberForm);
+  const isOpenTeamNewMemberForm = useSelector(selectIsOpenNewTeamMemberForm);
   const employeesSelectOptions = useSelector(selectEmployeesSelectOptions);
+  const teamNewTeamMemberFormErrors = useSelector(
```
Comment (kwoon) [inline js/src/pages/ProjectTeamPage.tsx:98]: Typo, teamNewTeamMemberFormErrors => newTeamMemberFormErrors

File: js/src/features/project/epics.ts [hunk 13]
```diff
@@ -171,7 +173,7 @@ export const fetchProjectRolePermissionsEpic: RootEpic = (
         map((result) =>
           result.success
             ? fetchProjectRoleAsync.success(result.data)
-            : fetchProjectRoleAsync.failure(result.errors),
+            : fetchProjectRoleAsync.failure(result.reason),
```
Comment (kwoon) [inline js/src/features/project/epics.ts:176]: You cannot just output reason. You should handle it and output your interpritation for this reason.
Again, `reason` - it is just constant.

File: js/src/features/notification/epics.ts [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+import { notification } from "antd";
+
+import { RootEpic } from "MyTypes";
+import { notificationError, notificationSuccess } from "./actions";
+
+export const notificationSuccessEpic: RootEpic = (action$) =>
+  action$.pipe(
+    filter(isActionOf(notificationSuccess)),
+    tap((action) => {
+      console.log("test");
```
[the remark this answers is not in the document]
Comment (Aliaksandrzaz) [reply js/src/features/notification/epics.ts:12]: Fixed

File: js/src/features/notification/epics.ts [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+      notification.success({ message: action.payload });
+    }),
+    ignoreElements(),
+  );
+
+export const notificationFailedEpic: RootEpic = (action$) =>
+  action$.pipe(
+    filter(isActionOf(notificationError)),
+    tap((action) => {
+      console.log(action.payload);
```
[the remark this answers is not in the document]
Comment (Aliaksandrzaz) [reply js/src/features/notification/epics.ts:22]: Fixed

File: js/src/pages/ProjectTeamsPage.tsx (hunk 10 above)
Comment (Aliaksandrzaz) [reply js/src/pages/ProjectTeamsPage.tsx:76]: Done

File: js/src/pages/ProjectTeamsPage.tsx (hunk 11 above)
Comment (Aliaksandrzaz) [reply js/src/pages/ProjectTeamsPage.tsx:50]: Done

File: js/src/pages/ProjectTeamPage.tsx (hunk 12 above)
Comment (Aliaksandrzaz) [reply js/src/pages/ProjectTeamPage.tsx:98]: Fixed

File: js/src/pages/ProjectsPage.tsx (hunk 9 above)
Comment (Aliaksandrzaz) [reply js/src/pages/ProjectsPage.tsx:47]: Done

File: js/src/features/project/components/Role/RoleFormTableWrapper.tsx (hunk 4 above)
Comment (Aliaksandrzaz) [reply js/src/features/project/components/Role/RoleFormTableWrapper.tsx:7]: Done

File: js/src/pages/ProjectsPage.tsx (hunk 8 above)
Comment (Aliaksandrzaz) [reply js/src/pages/ProjectsPage.tsx:19]: Done

File: js/src/features/project/epics.ts (hunk 5 above)
Comment (Aliaksandrzaz) [reply js/src/features/project/epics.ts:28]: Done

File: js/src/features/notification/actions.ts [hunk 16]
```diff
@@ -0,0 +1,9 @@
+import { createAction } from "typesafe-actions";
+
+export const notificationSuccess = createAction("Notification success")<
```
[the remark this answers is not in the document]
Comment (Aliaksandrzaz) [reply js/src/features/notification/actions.ts:3]: Done

File: js/src/features/project/components/Project/ProjectSlugForm.tsx (hunk 2 above)
Comment (Aliaksandrzaz) [reply js/src/features/project/components/Project/ProjectSlugForm.tsx:50]: Done

File: js/src/features/project/components/Project/ProjectCreateForm.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Aliaksandrzaz) [reply js/src/features/project/components/Project/ProjectCreateForm.tsx:53]: Done

File: js/src/features/project/components/Project/ProjectSlugForm.tsx (hunk 3 above)
Comment (Aliaksandrzaz) [reply js/src/features/project/components/Project/ProjectSlugForm.tsx:52]: Done

File: js/src/services/team-api.ts (hunk 6 above)
Comment (Aliaksandrzaz) [reply js/src/services/team-api.ts:83]: Done

File: js/src/services/project-api.ts (hunk 7 above)
Comment (Aliaksandrzaz) [reply js/src/services/project-api.ts:34]: Done

**Pull request merged**